### PR TITLE
Removes configure, Improves dict_merge, Adds merging unittest, Fixes Lookup unittests

### LINKED ISSUE #45: Remove configure module?
We need to decide whether to use the configure module or not.
Currently, we're using the module to create 'Configuration' (dict like) instances that load and merge yaml files and to register constructor for yaml tags.
If we decide to use this module, we can delete the 'cli.utils.dict_merge' method because the Configuration class already implements a 'merge' method

### PR DESCRIPTION
**Pull request by aopincar**
DO NOT MERGE YET!!!
In the 'internal' input file we use, there is a use of the old '!env' & '!join' tags.
With this patch we can't use them, so don't while we make use of the 'internal' settings file.

Resolves #45

### CODE REVIEW

File: cli/utils.py [hunk 1]
```diff
@@ -62,8 +64,7 @@ def validate_settings_dir(settings_dir=None):
     :raise: IRFileNotFoundException: when the path to the settings dir doesn't
             exist
     """
-    settings_dir = settings_dir or os.environ.get(
-        utils.INFRARED_DIR_ENV_VAR)
+    settings_dir = settings_dir or os.environ.get(INFRARED_DIR_ENV_VAR)
```
Comment (yfried-redhat) [inline cli/utils.py]: This method should be part of conf module, and not utils

File: cli/yamls.py [hunk 2]
```diff
@@ -15,95 +13,6 @@
 
 LOG = logger.LOG
```
Comment (yfried-redhat) [inline cli/yamls.py]: So, This removes the metaclass, and this module no longer needs to be imported on init, does it?
Also, consider renaming this module, or merging it with utils

**Comment by gamado** [comment]
Can one of the admins verify this patch?

**Comment by rhosqeauto** [comment]
Can one of the admins verify this patch?

**Comment by yfried-redhat** [comment]
retest this please

**Comment by gamado** [comment]
retest this please

**Comment by gamado** [comment]
retest this please

**Comment by yfried-redhat** [comment]
@aopincar Please add this to our JIRA if it's still relevant. Otherwise, please close this

**Comment by gamado** [comment]
retest

**Comment by aopincar** [comment]
Implemented in #156

**Pull request closed without merging**